### PR TITLE
Implementation of the rotation of the board view

### PR DESCRIPTION
**Pull request by Funkenschlag1**
Hi! I implemented the ability to rotate the board like requested in issue  [#297](https://github.com/sanderland/katrain/issues/297). It also changes the coordinates and works with edge cases like non-square boards and resizing the window while the board is rotated.

### CODE REVIEW

File: katrain/gui/badukpan.py [hunk 1]
```diff
@@ -253,91 +263,239 @@ def draw_board(self, *_args):
...
+
+        return extra_px_margin_x, extra_px_margin_y
+
+
+    def calculate_stone_size(self, grid_size):
+        return grid_size * Theme.STONE_SIZE
+
+
+    def initialize_gridpos(self):
+        self.gridpos = np.zeros((len(self.initial_gridpos_y), len(self.initial_gridpos_x)), dtype=object)
```
Comment (sanderland) [inline katrain/gui/badukpan.py:350]: any reason this isn't np.zeros(x,y,2) ?

File: katrain/gui/badukpan.py [hunk 2]
```diff
@@ -253,91 +263,239 @@ def draw_board(self, *_args):
...
+        for y in range(len(self.initial_gridpos_y)):
+            for x in range(len(self.initial_gridpos_x)):
+                self.gridpos[y][x] = (self.initial_gridpos_x[x], self.initial_gridpos_y[y])
+
+
+    def initialize_gridpos_x_y(self, board_size_x, board_size_y, grid_spaces_margin_x, grid_spaces_margin_y, extra_px_margin_x, extra_px_margin_y, grid_size):
```
Comment (sanderland) [inline katrain/gui/badukpan.py:356]: combine with initialize_gridpos?

**Review by sanderland — COMMENTED** [review]
some minor comments, still working out details

File: katrain/gui/badukpan.py (hunk 2 above)
Comment (Funkenschlag1) [reply katrain/gui/badukpan.py:356]: I reuse initialize_gridpos_x_y in calculate_rotated_gridpos but not initialize_gridpos, so it makes sense to keep them separate I think

**Pull request merged**